awx/api/serializers.py

```diff
@@ -4453,6 +4577,335 @@ def validate(self, attrs):
+    credentials = serializers.ListField(child=serializers.IntegerField(min_value=1), required=False)
+    identifier = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    labels = serializers.ListField(child=serializers.IntegerField(min_value=1), required=False)
+    instance_groups = serializers.ListField(child=serializers.IntegerField(min_value=1), required=False)
+    execution_environment = serializers.IntegerField(required=False, min_value=1)
+    #
+    limit = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    scm_branch = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    verbosity = serializers.IntegerField(required=False, min_value=1)
+    forks = serializers.IntegerField(required=False, min_value=1)
+    char_prompts = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    diff_mode = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    job_tags = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    job_type = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    skip_tags = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    survey_passwords = serializers.CharField(required=False, write_only=True, allow_blank=False)
+    job_slice_count = serializers.IntegerField(required=False, min_value=1)
+    timeout = serializers.IntegerField(required=False, min_value=1)
+
+    class Meta:
+        fields = (
+            'unified_job_template',
+            'identifier',
+            'inventory',
+            'credentials',
+            'limit',
+            'labels',
+            'instance_groups',
+            'execution_environment',
+            'scm_branch',
+            'verbosity',
+            'forks',
+            'char_prompts',
+            'diff_mode',
+            'extra_data',
+            'job_slice_count',
+            'job_tags',
+            'job_type',
+            'skip_tags',
+            'survey_passwords',
+            'timeout',
+            # these are related objects and we need to add extra validation for them in the parent BulkJobLaunchSerializer
+            #
+        )
+
+
+class BulkJobLaunchSerializer(BaseSerializer):
+    name = serializers.CharField(default='Bulk Job Launch', max_length=512, write_only=True, required=False, allow_blank=True)  # limited by max name of jobs
+    job_node_help_text = _('List of jobs to be launched, JSON. e.g. [{"unified_job_template": 7}, {"unified_job_template": 10}]')
+    jobs = BulkJobNodeSerializer(many=True, allow_empty=False, write_only=True, max_length=1000, help_text=job_node_help_text)
+    description = serializers.CharField(write_only=True, required=False, allow_blank=False)
+    extra_vars = serializers.CharField(write_only=True, required=False, allow_blank=False)
+    organization = serializers.PrimaryKeyRelatedField(
```

Comment:
I am confused what `organization` does, and to a lesser extent, name too.

Normally, when you launch or update something (a UJT, unified job template), the update/job inherits the name from the UJT, and inherits the organization from the UJT.

You could give something job a name unrelated to its template, and this may make it hard for the user to _find_ and understand the jobs, but there's no fundamental problem. But for `organization`, we depend on organization being the same between the job & the UJT for permissions to be coherent.

Reply:
@AlanCoding Right, there is no ujt for a bulk job. We are not using any ujt, we creating the workflow job directly, which is why we cant inherit the orgs and name from a ujt. We are assigning them directly.

Reply:
do you have some docs that describe the current form of the feature?

Reply:
Not yet, but let me add some docs.

Reply:
added some docs for the feature.